lib/components/link/link.less

```diff
@@ -1,10 +1,26 @@
 // TODO we *really* shouldn't be apply styles directly onto `<a>` like this, but
 // it's tech debt that'll take some doing in consumer's code to pay down.
+a {
+    &:visited {
+        &:not([class*="s-"]),
```

Comment:
So in here we are saying that if an anchor element has a stacks class which is not `s-link`, `s-sidebarwidget--action` or `s-user-card--link` should not get the visited styles. Out of curiosity why is this necessary? What happen if we would just target all the anchor tags (including the ones that have stacks classes applied)?

Regardless this line might be worth a small comment.

Reply:
> Out of curiosity why is this necessary? 

To make sure we're on the same page, this ensures we only set visited styles on:

- `a.s-link`
- `a.s-sidebarwidget--action`
- `a.s-user-card--link`
- all other anchors `a` that don't include a stacks component class (identified by `.s-`)

These rules keep it (somewhat) narrowly scoped and only applies it to elements where we can be confident the visited style is appropriate.

> What happen if we would just target all the anchor tags (including the ones that have stacks classes applied)?

If we broadly targeted _all anchors_, this style could wind up being applied to components like `.s-btn` and `.s-tag`. If they don't have explicit `:visited` styles defined, this rule would potentially override any color set on the element in favor of the visited color, resulting in unexpected colors and possibly causing contrast issues.

For example, here's a visited moderator tag if we targeted `a:visited` here:

![image](https://github.com/user-attachments/assets/eb548fda-6433-4b16-8e18-2c6b917f33cc)


> Regardless this line might be worth a small comment.

This seems like a good call. When I came back to this PR, I had to take a moment to refresh myself on what why this rule was structured this way (basically, I had to ask myself the same questions you asked). I'll add a comment describing the reasoning.